olaaf_django/tests/conftest.py

```diff
@@ -0,0 +1,189 @@
+
+
+# branches as keys | commits and changed files as values
+PUBLICATION_BRANCHES = {
+    "publication/2020-01-01": {
+        "publication/2020-01-01": [],  # empty commit
+        "2020-01-01/2019-01-01": [(f.name, False) for f in ALL_FILES],  # 4 files * (B|R) = 8 hashes
+        "2020-01-01/2019-02-02": [  # 2 files * (B|R) = 4 hashes
+            # file name, indicator if content of the authenticated div is modified
+            ("file1.html", True),
+            ("file3.html", True)
+        ],
+        "2020-01-01/2019-03-03": [  # 1 file * (B|R) = 2 hashes
+            ("file2.html", True)
+        ],
+        "2020-01-01/2019-04-04": [  # 1 file * (B|R) = 2 hashes
+            ("index.html", True)
+        ],
+        "2020-01-01/2019-05-05": [  # 1 file * (B) = 1 hashes
+            ("file1.html", False)
+        ]
+    },
+    "publication/2020-05-05": {
+        "publication/2020-05-05": [],
+        "2020-05-05/2019-01-01": [(f.name, False) for f in ALL_FILES],
+        "2020-05-05/2019-02-02": [
+            ("file1.html", True),
+            ("file3.html", True)
+        ],
+        "2020-05-05/2019-03-03": [
+            ("file2.html", True)
+        ],
+        "2020-05-05/2019-04-04": [
+            ("index.html", True)
+        ],
+        "2020-05-05/2019-05-05": [
+            ("file1.html", False)
+        ],
+        "2020-05-05/2019-06-06": [
+            ("file2.html", False)
+        ],
+        "2020-05-05/2019-07-07": [
+            ("index.html", True)
+        ],
+    },
+    "publication/2020-05-05-01": {  # publication/2020-05-05  will be skipped
+        "publication/2020-05-05-01": [],
+        "2020-05-05/2019-01-01": [(f.name, False) for f in ALL_FILES],
+        "2020-05-05/2019-02-02": [
+            ("file1.html", True),
+            ("file3.html", True)
+        ],
+        "2020-05-05/2019-03-03": [
+            ("file2.html", True)
+        ],
+        "2020-05-05/2019-04-04": [
+            ("index.html", True)
+        ],
+        "2020-05-05/2019-05-05": [
+            ("file1.html", False)
+        ],
+        "2020-05-05/2019-06-06": [
+            ("file2.html", False)
+        ],
+        "2020-05-05/2019-07-07": [
+            ("index.html", True)
+        ],
+        "2020-05-05/2019-08-08": [
+            ("index.html", True)
+        ],
+        "2020-05-05/2019-09-09": [
+            ("file2.html", True)
+        ],
+        "2020-05-05/2019-10-10": [
+            ("file3.html", True)
+        ],
+    },
+}
+
+NON_PUBLICATION_BRANCHES = {
+    "publication/2020-01": {},
+    "publication/2020-01-01-01-01": {},
+    "publication/2020-01-01.2020-01-04": {},
+    "test": {},
+}
+
+
+@pytest.fixture(scope='session')
+def repository():
+  try:
+    repo = GitRepository(str(REPOSITORY_PATH))
+    repo.init_repo()
+    _init_pub_branches(repo)
+    yield repo
+  except KeyboardInterrupt:
+    pass
+  except Exception:
+    raise
+  finally:
+    shutil.rmtree(REPOSITORY_PATH)
```

Comment:
This raises an Access denied exception on Windows. Add `onerror`. We use the following implementation in TAF:
```
def onerror(_func, path, _exc_info):
    """Used by when calling rmtree to ensure that readonly files and folders
    are deleted.
    """
    os.chmod(path, stat.S_IWRITE)
    os.unlink(path)
```
Tested and works here too